ui/segmented-bar.reel/segmented-bar.js

```diff
@@ -0,0 +1,259 @@
+/**
+ * @module montage/ui/segmented-bar.reel
+ * @requires montage/ui/base/abstract-button.AbstractButton
+ * @requires montage/core/tree-controller.TreeController
+ */
+var AbstractButton = require("ui/base/abstract-button").AbstractButton;
+
+/**
+ * @class SegmentedBar
+ * @classdesc
+ * A segmented bar.
+ * Reference design: OS X storage space visualization found in iTunes or About This Mac
+ * Inheriting from AbstractButton because SegmentedBar needs its PressComposer behavior
+ *
+ * @extends AbstractButton
+ *
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Flexible_boxes}
+ * for FlexBox usage
+ */
+exports.SegmentedBar = AbstractButton.specialize( /** @lends SegmentedBar.prototype # */ {
```

Comment:
Duh. Sorry about that.

   - C=

On Fri, Feb 20, 2015 at 1:12 PM, Ray Shan <notifications@github.com> wrote:

> In ui/segmented-bar.reel/segmented-bar.js
> <https://github.com/montagejs/montage/pull/1515#discussion_r25094502>:
>
> > + */
> > +var AbstractButton = require("ui/base/abstract-button").AbstractButton;
> > +
> > +/**
> > + * @class SegmentedBar
> > + * @classdesc
> > + * A segmented bar.
> > + * Reference design: OS X storage space visualization found in iTunes or About This Mac
> > + * Inheriting from AbstractButton because SegmentedBar needs its PressComposer behavior
> > + *
> > + * @extends AbstractButton
> > + *
> > + * @see {@link https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Flexible_boxes}
> > + * for FlexBox usage
> > + */
> > +exports.SegmentedBar = AbstractButton.specialize( /** @lends SegmentedBar.prototype # */ {
>
>  line 138: Why isn't this done through a binding in the serialization?
>
> b/c why spend resources on a binding data if user is just passing a
> dataController
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/montagejs/montage/pull/1515/files#r25094502>.
>